ocrd_utils/ocrd_utils/os.py

```diff
@@ -121,16 +121,19 @@ def atomic_write(fpath):
     with atomic_write_(fpath, writer_cls=AtomicWriterPerms, overwrite=True) as f:
         yield f
 
-def resolve_mets_arguments(directory, mets_url, mets_basename, log):
+def resolve_mets_arguments(directory, mets_url, mets_basename, log=None):
     """
     Resolve the ``--mets``, ``--mets-basename`` and `--directory`` argument
     into a coherent set of arguments according to https://github.com/OCR-D/core/issues/517
     """
     if mets_basename and mets_url:
         raise ValueError("Use either --mets or --mets-basename, not both")
     if mets_basename and not mets_url:
-        log.warning(DeprecationWarning("--mets-basename is deprecated. Use --mets/--directory instead"))
-    mets_basename = mets_basename if mets_basename else 'mets.xml'
+        (log.warning if log else print)(DeprecationWarning("--mets-basename is deprecated. Use --mets/--directory instead"))
```

Comment:
Wow, that's almost Lisp syntax. I didn't know Python is that expressive.

Reply:
Lispier than https://github.com/OCR-D/core/blob/master/ocrd/ocrd/workspace.py#L573-L580 ? :D

```python
               if (all(feature in featureset
                        for feature in feature_selector.split(',') if feature) and
                    not any(feature in featureset
                            for feature in feature_filter.split(',') if feature) and
                    len(featureset.difference(auto_features)) >= \
                    len(best_features.difference(auto_features))):
                    best_features = featureset
                    best_image = alternative_image
```

Reply:
But that's just predicating on a list of booleans. Yours is predicating on predicates (higher order). I guess subconsciously I was aware functions are first class citizens in Python, too. I just didn't know you can use the the parentheses syntax of standard lambda calculus for that. (In many other places, indentation is used instead of s-exp syntax...)